src/importer/sources/retroarch_source.py

```diff
@@ -0,0 +1,119 @@
+# retroarch_source.py
+#
+# Copyright 2023 Rilic
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from pathlib import Path
+from time import time
+
+import json
+import os
+import logging
+from json import JSONDecodeError
+
+from src import shared
+from src.game import Game
+from src.importer.sources.location import Location
+from src.importer.sources.source import (
+    SourceIterationResult,
+    SourceIterator,
+    Source
+)
+
+
+class RetroarchSourceIterator(SourceIterator):
+    source: "RetroarchSource"
+
+    def generator_builder(self) -> SourceIterationResult:
+        playlist_files = []
+        for file in os.listdir(self.source.config_location["playlists"]):
+            if file.endswith('.lpl'):
+                playlist_files.append(file)
+
+        playlist_items = []
+        for playlist_file in playlist_files:
+            open_file = open(str(self.source.config_location["playlists"]) + "/" + playlist_file)
```

Comment:
Use `with open(...) as open_file` inside of the `try` block, and don't forget to specify the encoding ;)